src/python/gem5/simulate/exit_event_generators.py

```diff
@@ -143,3 +144,24 @@ def save_checkpoint_generator(checkpoint_dir: Path):
     while True:
         m5.checkpoint((checkpoint_dir / f"cpt.{str(m5.curTick())}").as_posix())
         yield False
+
+def simpoints_save_checkpoint_generator(
+    checkpoint_dir: Path,
+    simpoint: SimPoint
+    ):
+    simpoint_list = simpoint.get_simpoint_start_insts()
+    count = 0
+    last_start = -1
+    while True:
+        m5.checkpoint((checkpoint_dir / f"cpt.SimPoint{count}").as_posix())
+        last_start = simpoint_list[count]
+        count += 1
+        while(count < len(simpoint_list) and 
```

Comment:
Can you add a comment to explain why this while loop is needed?

Reply:
Added comments :)